backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ProjectDatabaseHandler.java

```diff
@@ -154,6 +156,86 @@ public ProjectDatabaseHandler(Supplier<HttpClient> httpClient, String dbName, St
         return repository.searchByName(name, user);
     }
 
+    /////////////////////////////
+    // CREATE CLEARING REQUEST //
+    /////////////////////////////
+
+    public AddDocumentRequestSummary createClearingRequest(ClearingRequest clearingRequest, User user, String projectUrl) throws SW360Exception {
+        Project project = getProjectById(clearingRequest.getProjectId(), user);
+        AddDocumentRequestSummary requestSummary = new AddDocumentRequestSummary().setRequestStatus(AddDocumentRequestStatus.FAILURE);
+
+        if (!(PermissionUtils.isUserAtLeast(UserGroup.CLEARING_ADMIN, user) || isWriteActionAllowedOnProject(project, user))) {
+            return requestSummary.setMessage("You do not have WRITE access to the project!");
+        }
+
+        if (CommonUtils.isNotNullEmptyOrWhitespace(project.getClearingRequestId())) {
+            log.error("Clearing request is already present for the project: " + project.getId());
+            return requestSummary.setRequestStatus(AddDocumentRequestStatus.DUPLICATE).setId(project.getClearingRequestId());
+        }
+
+        if (!SW360Utils.isValidDate(clearingRequest.getRequestedClearingDate(), DateTimeFormatter.ISO_LOCAL_DATE, 7)) {
+            log.error("Invalid requested clearing date: " + clearingRequest.getRequestedClearingDate() + " is entered, by user: "+ user.getEmail());
+            return requestSummary.setMessage("Invalid requested clearing date!");
+        }
+
+        if (!(ProjectClearingState.CLOSED.equals(project.getClearingState()) || Visibility.PRIVATE.equals(project.getVisbility()))) {
+            clearingRequest.setProjectBU(project.getBusinessUnit());
+            String crId = moderator.createClearingRequest(clearingRequest, user);
+            if (CommonUtils.isNotNullEmptyOrWhitespace(crId)) {
+                project.setClearingRequestId(crId);
+                repository.update(project);
```

Comment:
The requirement states:

>user with role CLEARING_ADMIN and above are allowed to create clearing request.

So, even if `Clearing Admin` or `Admin` doesn't have `WRITE` access to project, They should be able to create the clearing request. which is not possible if I use existing `updateProject()` method, as it will create a `Moderation Request` if user doesn't have `WRITE` access.

project is updated at 2 scenarios:
* When creating the `CR` (to add `CR` Id in project document.)
* Whenever there is a change in `CR` state, which in-turn changes the project `clearing state`.

@mcjaeger Can you please suggest what would be the best approach for handling this scenario.